### PR TITLE
[Guided Setup: Stage 3] Send the Onboarding Message After Sign-In

### PR DESCRIPTION
**Pull request by cdOut**
### Details

Added logic for sending onboarding messages through concierge after sign-in through a magic link sent via email for multiple cases.

### Fixed Issues

$ https://github.com/Expensify/App/issues/45044
PROPOSAL:

### Workspace Non-admin:

1. On the first account create or use an existing worspace
2. Add a member using an email add an account that you know doesn't exist yet
3. Make sure the account you've invited is set as just a Member
4. Proceed to your email and press on the link on the bottom of the invitation email you've received
5. Confirm that that a Concierge message appears and it shows the correct content, as well as an onboarding video.

### Workspace Admin:

1. On the first account create or use an existing worspace
2. Add a member using an email add an account that you know doesn't exist yet
3. Press on the newly invited user and make sure that his role is set to Admin
4. Proceed to your email and press on the link on the bottom of the invitation email you've received
5. Confirm that that a Concierge message appears and it shows the correct content, as well as an onboarding video.

### DM / Group Chat:

1. On the first account create or use an existing DM or Group Chat
2. Add a member using an email add an account that you know doesn't exist yet, in the case of the room you can add by mentioning the user
4. Proceed to your email and press on the link on the bottom of the invitation email you've received
5. Confirm that that a Concierge message appears and it shows the correct content, as well as an onboarding video. Make sure the ~Start a chat~ is checked off & struck through.

### User-created Room:

1. On the first account create or use an existing user-created room
2. Add a member using an email add an account that you know doesn't exist yet, do it by mentioning the gmail in the room chat
4. Proceed to your email and press on the link on the bottom of the invitation email you've received
5. Confirm that that a Concierge message appears and it shows the correct content, as well as an onboarding video.

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos

<details>
<summary>MacOS: Chrome / Safari</summary>

#### Workspace Admin

https://github.com/user-attachments/assets/66a564cb-6465-4f82-b190-ea4b8c740d2e

#### Workspace Non-Admin

https://github.com/user-attachments/assets/91743bce-44af-4d89-9846-7e379c271391

</details>

### CODE REVIEW

**Comment by trjExpensify** [comment]
👋 @getusha can you review this one today, please? Thanks!

**Comment by getusha** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/CONST.ts
Comment (getusha) [inline src/CONST.ts:4300]: Why is this?

File: src/libs/actions/Report.ts
Comment (getusha) [inline src/libs/actions/Report.ts:780]: NAB:
```suggestion
    if (choice === CONST.ONBOARDING_CHOICES.ADMIN || choice === CONST.ONBOARDING_CHOICES.SUBMIT) {
            return {
                firstName,
                lastName,
                onboardingMessage: CONST.ONBOARDING_MESSAGES[choice],
            };
    }
```

File: src/libs/actions/Report.ts
Comment (getusha) [inline src/libs/actions/Report.ts:759]: NAB:
```suggestion
    if (isInviteOnboardingComplete || inviteType === CONST.ONBOARDING_INVITE_TYPES.IOU || inviteType === CONST.ONBOARDING_INVITE_TYPES.INVOICE) {
        return;
```

File: src/libs/actions/Report.ts
Comment (getusha) [inline src/libs/actions/Report.ts:745]: I don't think we want to keep this, correct?

File: src/libs/actions/Report.ts
Comment (getusha) [inline src/libs/actions/Report.ts:748]:
```suggestion
 * Gets details for an invite onboarding if certain conditions are met
 * @returns firstName, lastName, and onboardingMessage values
```

**Comment by getusha** [comment]
Not able to get an invitation email after inviting new member to a workspace, is it working for you? @cdOut

File: src/libs/actions/Report.ts
Comment (cdOut) [reply src/libs/actions/Report.ts]: Good catch, I've cleared it up.

File: src/libs/actions/Report.ts
Comment (cdOut) [reply src/libs/actions/Report.ts]: Cleared this one also.

File: src/libs/actions/Report.ts
Comment (cdOut) [reply src/libs/actions/Report.ts]: We don't, I overlooked removing this after writing up the proper comment of this function somehow.

File: src/libs/actions/Report.ts
Comment (cdOut) [reply src/libs/actions/Report.ts:736]: Added.

File: src/CONST.ts
Comment (cdOut) [reply src/CONST.ts:4300]: @kosmydel who originally worked on this wanted to have better typescript checks for the message types so this has been removed in favor of adding a `satisfies Record<OnboardingPurposeType, OnboardingMessageType>` to the whole message CONST definition just to be clear that the values defined there are correct.

**Comment by cdOut** [comment]
@getusha I've cleaned up the code after your review.

I did not encounter an issue with receiving the invitation emails, no. Did you check after by logging into the invited account if they are even being properly added to said workspace?

**Comment by trjExpensify** [comment]
@getusha let me know what email address you used to invite and invited.

**Comment by getusha** [comment]
>Automated Tests

@cdOut Is this planned to be worked on separately? https://github.com/Expensify/App/issues/45044

And lets make sure we are covering the cases in the feature description in test steps.

**Comment by cdOut** [comment]
> > Automated Tests
> @cdOut Is this planned to be worked on separately? #45044

@deetergp should the automated tests for it be added here or handled separately?

> And lets make sure we are covering the cases in the feature description in test steps.

Also I'm fairly sure all the cases are handled in test steps, I've separated them into three different sections.

File: src/CONST.ts
Comment (blazejkustra) [inline src/CONST.ts:130]: Let's not add suffixes as 'Type' 'Interface' 'I' to the names of types (it's in our STYLE.md guidelines somewhere) 😅

Comment (blazejkustra) [reply src/CONST.ts:130]: Instead let's use `OnboardingPurpose` or `OnboardingChoice`

File: src/CONST.ts
Comment (blazejkustra) [inline src/CONST.ts:138]: This is different and Type suffix can stay. Here, `Type` indicates different categories of `OnboardingInvite`, thus adding significant context to the definition.

File: src/CONST.ts
Comment (blazejkustra) [inline src/CONST.ts:140]: Here it's wrong again, `OnboardingTask` is better

File: src/CONST.ts
Comment (blazejkustra) [inline src/CONST.ts:147]:
```suggestion
type OnboardingMessage = {
```

File: src/CONST.ts
Comment (blazejkustra) [inline src/CONST.ts:151]: Let's add JSDoc to these, wdyt?

File: src/ONYXKEYS.ts
Comment (blazejkustra) [inline src/ONYXKEYS.ts:3]: I think it would be better if we move `OnboardingPurposeType` to `types/onyx`, don't you think?

File: src/types/onyx/IntroSelected.ts
Comment (blazejkustra) [inline src/types/onyx/IntroSelected.ts:4]: Why do we need to use partial here?

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/types/onyx/IntroSelected.ts
Comment (cdOut) [reply src/types/onyx/IntroSelected.ts:4]: With the new non-organic flow there are cases where the object returned from `NVP_INTRO_SELECTED` is empty, which is why we've changed it to a `Partial` here

**Review by blazejkustra — APPROVED** [review]

File: src/types/onyx/IntroSelected.ts
Comment (cdOut) [reply src/types/onyx/IntroSelected.ts:5]: As you've suggested I've removed the `Partial` here and replaced it with optional parameters for better TS performance

**Comment by blazejkustra** [comment]
@getusha All yours!

**Comment by cdOut** [comment]
Please hold off on the review just yet - the code fully works but after some [clarifications with Scott](https://swmansion.slack.com/archives/C036QM0SLJK/p1725490576502609?thread_ts=1725284010.178449&cid=C036QM0SLJK) I had to rework it to send all the onyx data for concierge messages and tasks inside of `openReport` instead of utilizing the `completeOnboarding` function and API call. I'll make sure to notify you when it's ready.

**Comment by cdOut** [comment]
@getusha there’s still this issue we’re trying to resolve but the rest is working, we think it’s best for you to start the review while me and Scott work on fixing the autocomplete issue to move this one along.

**Comment by getusha** [comment]
Testing

**Comment by getusha** [comment]
The link i receive via email doesn't work most of the time. any idea @cdOut?

https://github.com/user-attachments/assets/ec02aa5b-3766-41a6-b50e-1282d027d4fa

**Comment by cdOut** [comment]
> The link i receive via email doesn't work most of the time. any idea @cdOut?

What do you mean by that? It appears to work as expected in the video you've attached. If you aren't receiving the invitation emails or they are very delayed, then it's due to the backend and I am unable to fix that. You could always try swtiching to the staging backend servers.

**Comment by getusha** [comment]
Sorry, attached a wrong video. this is the link.

`https://email.expensify.com/c/eJxUzj1v8jAQAOBfc9ledD47djJ44AW5Q7-ktlMX5CRncEVilFyA_vtK3bo_wzP41lmLFXvlyFGNRqnq5LVLjU7ExpKhruvjEFNnmNoY2ekhVdkTksGWanTG1mrDUWvSQ9v0aN2gEAzy_cLTktP3pi9jdfYnkcsCegsUgMLEt80fARSuQEE12qGyNVDYPu8fXx-etp-gA9-zfBTQe6AwAwVCdI2m1lpslXKumv2RZR14ybIC_Z_49i8aMHgcYz7_DsS_FMkp91FymQ5vfCmzHHbrImV8X7sv7mVXxpEnqa6efgIAAP__DjdVUg`

I think it should've been something like this to work:
`https://new.expensify.com/v/18370080/CNZAQKIJK?exitTo=/r/7302026878763812`

**Comment by cdOut** [comment]
> This PR hasn't touched emails has it? 🤔

It has not, but for whatever reason we send two types of links when inviting new users. One is in the form of `https://new.expensify / https://staging.new.expensify` and `https://email.expensify`. I'm not sure why that is, @deetergp could probably answer that one, but they essentially redirect to the same flow in the app. The only problem is that the `email `variant isn't modifiable to test on the dev branch (as in swapping` staging.new.expensify` to `dev.new.expensify.com:8082` for example) so that's why it's problematic here.

Still, this doesn't block testing and at best is just a slight inconvenience. My advice would be to swap to use staging backend for testing and just retry until you get a valid replacable url for local dev.

**Comment by getusha** [comment]
@cdOut I signed in via an invite link on Android native, but the video isn't loading. Is this something we can fix here?
1. Invite a user that doesn't exist yet
2. Change the role to admin
3. Open the invite link on android  `npx uri-scheme open "new-expensify://v/18376702/VDLKQYZTW\?exitTo\=/r/3948290673863711" --android` or just directly sign-in using the email address

<image src="https://github.com/user-attachments/assets/2bedbc93-e5f1-49c6-b1c0-44485b311323" width="350" />
<image src="https://github.com/user-attachments/assets/79944396-66c3-4c5d-bb5f-31d422b92816" width="350" />

**Comment by trjExpensify** [comment]
If you organically sign-up and choose "Manage my team's expenses" - does the video load in that flow on Android native?

**Comment by getusha** [comment]
>If you organically sign-up and choose "Manage my team's expenses" - does the video load in that flow on Android native?

Yes

<image src="https://github.com/user-attachments/assets/a2e8cfa0-a471-4475-abbc-de377e5fb70c" width="350" />

**Comment by cdOut** [comment]
> @cdOut I signed in via an invite link on Android native, but the video isn't loading. Is this something we can fix here?

Would you be able to get the cURL and requestID so Scott could check what kinda data is being sent back over to you?

**Comment by deetergp** [comment]
@cdOut Do you happen to know which flows send which URLs? I'm not sure what those `email.` URLs are about.

@getusha Is the video failing to load only on Android? Does it show in that same flow on other platforms?

**Comment by getusha** [comment]
>@getusha Is the video failing to load only on Android? Does it show in that same flow on other platforms?

I was also able to reproduce on iOS. i'll check on other platforms as well.

**Comment by cdOut** [comment]
@getusha is that happening for one specific case or just in general? What are the steps to reproduce?

**Comment by getusha** [comment]
@cdOut happens for if the account role is set to admin.
here are the steps https://github.com/Expensify/App/pull/46052#issuecomment-2377108375

**Comment by getusha** [comment]
>@getusha could you specify if this only happens on android and check whether the specific flow you used on android works for desktop / web or does it also return broken video? Also try completing the tasks, do they complete successfully when pressing the checkbox or are they also broken? Is that just a problem with the video?

Works well on web, and it seems like it's broken it failed several times after checking the tasks off.

https://github.com/user-attachments/assets/5dd59874-e7e7-4a44-9440-c0e7235b9953

**Comment by cdOut** [comment]
@getusha Could you retrieve the `requestID` of `openReport` call which sends over the `guidedSetupData`? And is the `LateOnyxUpdate` from API returned correctly?

**Comment by getusha** [comment]
@cdOut 

`
 DEBUG  [info] [Network] Finished API request in 528ms - {"command":"OpenReport","jsonCode":200,"requestID":"8ccce7342d46973c-FRA"} {"request": {"command": "OpenReport", "cursorID": "", "data": {"accountIDList": "", "apiRequestType": "write", "canCancel": true, "clientLastReadTime": "2024-10-03 12:23:04.406", "emailList": "", "parentReportActionID": "-1", "pusherSocketID": "783029.79018", "reportActionID": "", "reportID": "94145616603358", "shouldRetry": true}, "failureData": [[Object]], "isPaginated": true, "resourceID": "94145616603358", "successData": [[Object], [Object]]}, "response": {"hasOlderActions": false, "jsonCode": 200, "onyxData": [[Object], [Object], [Object]], "requestID": "8ccce7342d46973c-FRA"}}
`

**Comment by getusha** [comment]
@cdOut are you not able to reproduce it?

**Comment by cdOut** [comment]
@getusha all the issues we've had for the backend were resolved today, I've tested it on my end and it runs properly for all cases. Ready for your *hopefully final* re-test!

**Comment by cdOut** [comment]
> this is the only message i am seeing from concierge

Alright let me take a look at it, any specific case?

**Comment by zfurtak** [comment]
hi @getusha, could you give a more specific scenario in which the issue occurred? 😊 So I can take care of this immediately, as @cdOut is ooo today and tomorrow 🔍

**Comment by cdOut** [comment]
I've asked about the issue in [this thread](https://swmansion.slack.com/archives/C07HPDRELLD/p1730132237694049?thread_ts=1729278293.829049&cid=C07HPDRELLD) on slack, seems like it's a backend issue once again as we don't have any way of modifying the email messages on frontend.

**Comment by blazejkustra** [comment]
> this is the only message i am seeing from concierge
> 
> Screenshot 2024-10-25 at 12 31 20 in the afternoon
> 
>  Screen.Recording.2024-10-25.at.12.20.58.in.the.afternoon.mov

@getusha I pushed new commits, can you retest ASAP? 🙏 (this PR has a CRITICAL priority now)

**Comment by deetergp** [comment]
> This is probably a backend bug, i am always getting expired links for User-created Room

Yeah, I'm getting that too, but I think we can safely consider it an unrelated bug, since if we sign in with the new user and request a new magic code, we do get the onboarding message and tasks.

**Comment by deetergp** [comment]
> I got duplicate messages a couple of times but not able to reproduce consistently, will update once i get clear repro steps.

I have not yet encountered that behavior, so I am interested to know what the reproduction steps are.

**Comment by blazejkustra** [comment]
> @blazejkustra, I was able to reproduce the issue. 

@zfurtak and I pair-programmed to identify the root cause. We discovered that sometimes two `OpenReport` requests are sent within a short timeframe. This, combined with the `checkAndFixConflictingRequest` functionality, caused the request containing the guided setup data to never be sent to the backend. The data was only provided again after a full page reload.

Last commit should finally resolve this issue once and for all. @getusha can you retest? 🙏

**Comment by deetergp** [comment]
OMGMGOMG are we going to merge this today?!

**Comment by deetergp** [comment]
@getusha That pesky console error keeps popping back up and is on the back end. Let's not hold on that.

**Comment by trjExpensify** [comment]
>@trjExpensify the Hey 👋 seems duplicate in the case of admin invitation, does it look fine to you?

Hm, it looks a bit weird. @danielrvidal I'll defer to you though.

**Comment by deetergp** [comment]
@getusha For DM & Group Chats, did you add a comment in either chat?

**Comment by danielrvidal** [comment]
@getusha good find. I think we should remove the "Hey ✋" at the beginning of the admin copy and just start with "As an admin..." 

Let's please make that update.

**Comment by deetergp** [comment]
@getusha To elaborate on my previous comment, for DMs & Group Chats, you might need to add a comment in the chat as the inviting user after inviting the new user, then wait for the normal "notify offline users of activity" time to elapse, which is about 10 minutes. They _should_ get sent immediately, but we have a regression that is preventing that from happening, but it shouldn't hold this PR up. If you do the above and still do not get an invitation, then that's a new regression… But they should send.

**Comment by zfurtak** [comment]
Hi, I updated the admin message and merged latest main. Are we good to merge or do we want to dig into above issues? 🔍

cc: @deetergp @getusha

**Comment by blazejkustra** [comment]
@getusha Could you finalize Reviewer Checklist? I think we are good to merge this 😄

**Review by getusha — APPROVED** [review]
LGTM, thanks!

**Review by deetergp — APPROVED** [review]
Woohoo! LGTM 🎉

**Comment by blazejkustra** [comment]
> This PR is failing because of issues #52540 and #52546

@IuliiaHerets This is a known issue, check the comment above: https://github.com/Expensify/App/pull/46052#issuecomment-2471323500

@deetergp I'm guessing it is a backend issue, so please handle it internally. If you need help on the FE side, let me know 🙇

**Comment by deetergp** [comment]
@IuliiaHerets Both issues are known, though I don't think we had GHs for them till now, so thanks for that. We will definitely want to add regression tests for both of them. But for now, use the workarounds from [the comment](https://github.com/Expensify/App/pull/46052#issuecomment-2471323500) @blazejkustra mentioned to verify that the onboarding messages are sent.

**Pull request merged**
